### ISSUE TITLE
Radical amount of redefinitions and unused variables in specs.

### ISSUE THREAD

**Issue by digitalextremist** [celluloid/celluloid#528]
```
/home/travis/build/celluloid/celluloid/lib/celluloid/rspec/actor_examples.rb:465: warning: assigned but unused variable - actor
/home/travis/build/celluloid/celluloid/lib/celluloid/rspec/actor_examples.rb:12: warning: method redefined; discarding old foo
/home/travis/build/celluloid/celluloid/lib/celluloid/rspec/actor_examples.rb:12: warning: method redefined; discarding old foo=
/home/travis/build/celluloid/celluloid/lib/celluloid/rspec/actor_examples.rb:990: warning: method redefined; discarding old subclass_proxy?
/home/travis/build/celluloid/celluloid/lib/celluloid/rspec/actor_examples.rb:990: warning: previous definition of subclass_proxy? was here
/home/travis/build/celluloid/celluloid/spec/celluloid/actor_system_spec.rb:20: warning: assigned but unused variable - thread
/home/travis/build/celluloid/celluloid/spec/celluloid/timer_spec.rb:36: warning: assigned but unused variable - times
```

and

```
/home/travis/build/celluloid/celluloid/lib/celluloid/properties.rb:10: warning: method redefined; discarding old mailbox_class
/home/travis/build/celluloid/celluloid/lib/celluloid/properties.rb:10: warning: previous definition of mailbox_class was here
/home/travis/build/celluloid/celluloid/lib/celluloid/properties.rb:10: warning: method redefined; discarding old proxy_class
/home/travis/build/celluloid/celluloid/lib/celluloid/properties.rb:10: warning: previous definition of proxy_class was here
/home/travis/build/celluloid/celluloid/lib/celluloid/properties.rb:10: warning: method redefined; discarding old task_class
/home/travis/build/celluloid/celluloid/lib/celluloid/properties.rb:10: warning: previous definition of task_class was here
/home/travis/build/celluloid/celluloid/lib/celluloid/properties.rb:10: warning: method redefined; discarding old mailbox_size
/home/travis/build/celluloid/celluloid/lib/celluloid/properties.rb:10: warning: previous definition of mailbox_size was here
/home/travis/build/celluloid/celluloid/lib/celluloid/properties.rb:10: warning: method redefined; discarding old exclusive_actor
/home/travis/build/celluloid/celluloid/lib/celluloid/properties.rb:10: warning: previous definition of exclusive_actor was here
/home/travis/build/celluloid/celluloid/lib/celluloid/properties.rb:10: warning: method redefined; discarding old exclusive_methods
/home/travis/build/celluloid/celluloid/lib/celluloid/properties.rb:10: warning: previous definition of exclusive_methods was here
/home/travis/build/celluloid/celluloid/lib/celluloid/properties.rb:10: warning: method redefined; discarding old execute_block_on_receiver
/home/travis/build/celluloid/celluloid/lib/celluloid/properties.rb:10: warning: previous definition of execute_block_on_receiver was here
/home/travis/build/celluloid/celluloid/lib/celluloid/properties.rb:10: warning: method redefined; discarding old finalizer
/home/travis/build/celluloid/celluloid/lib/celluloid/properties.rb:10: warning: previous definition of finalizer was here
/home/travis/build/celluloid/celluloid/lib/celluloid/properties.rb:10: warning: method redefined; discarding old exit_handler_name
/home/travis/build/celluloid/celluloid/lib/celluloid/properties.rb:10: warning: previous definition of exit_handler_name was here
/home/travis/build/celluloid/celluloid/lib/celluloid.rb:55: warning: method redefined; discarding old trap_exit
/home/travis/build/celluloid/celluloid/lib/celluloid.rb:55: warning: previous definition of trap_exit was here
/home/travis/build/celluloid/celluloid/lib/celluloid.rb:59: warning: method redefined; discarding old exclusive
/home/travis/build/celluloid/celluloid/lib/celluloid.rb:59: warning: previous definition of exclusive was here
/home/travis/build/celluloid/celluloid/lib/celluloid/rspec/actor_examples.rb:921: warning: method redefined; discarding old block
/home/travis/build/celluloid/celluloid/lib/celluloid/rspec/actor_examples.rb:921: warning: previous definition of block was here
/home/travis/build/celluloid/celluloid/lib/celluloid/rspec/actor_examples.rb:925: warning: method redefined; discarding old unblock
/home/travis/build/celluloid/celluloid/lib/celluloid/rspec/actor_examples.rb:925: warning: previous definition of unblock was here
```

and

```
/home/travis/build/celluloid/celluloid/spec/celluloid/supervision_group_spec.rb:60: warning: method redefined; discarding old args
/home/travis/build/celluloid/celluloid/spec/celluloid/supervision_group_spec.rb:61: warning: method redefined; discarding old initialize
/home/travis/build/celluloid/celluloid/spec/celluloid/supervision_group_spec.rb:7: warning: previous definition of initialize was here
```

and 

```
/home/travis/build/celluloid/celluloid/lib/celluloid/properties.rb:10: warning: method redefined; discarding old mailbox_class
/home/travis/build/celluloid/celluloid/lib/celluloid/properties.rb:10: warning: previous definition of mailbox_class was here
/home/travis/build/celluloid/celluloid/lib/celluloid/properties.rb:10: warning: method redefined; discarding old proxy_class
/home/travis/build/celluloid/celluloid/lib/celluloid/properties.rb:10: warning: previous definition of proxy_class was here
/home/travis/build/celluloid/celluloid/lib/celluloid/properties.rb:10: warning: method redefined; discarding old task_class
/home/travis/build/celluloid/celluloid/lib/celluloid/properties.rb:10: warning: previous definition of task_class was here
/home/travis/build/celluloid/celluloid/lib/celluloid/properties.rb:10: warning: method redefined; discarding old mailbox_size
/home/travis/build/celluloid/celluloid/lib/celluloid/properties.rb:10: warning: previous definition of mailbox_size was here
/home/travis/build/celluloid/celluloid/lib/celluloid/properties.rb:10: warning: method redefined; discarding old exclusive_actor
/home/travis/build/celluloid/celluloid/lib/celluloid/properties.rb:10: warning: previous definition of exclusive_actor was here
/home/travis/build/celluloid/celluloid/lib/celluloid/properties.rb:10: warning: method redefined; discarding old exclusive_methods
/home/travis/build/celluloid/celluloid/lib/celluloid/properties.rb:10: warning: previous definition of exclusive_methods was here
/home/travis/build/celluloid/celluloid/lib/celluloid/properties.rb:10: warning: method redefined; discarding old execute_block_on_receiver
/home/travis/build/celluloid/celluloid/lib/celluloid/properties.rb:10: warning: previous definition of execute_block_on_receiver was here
/home/travis/build/celluloid/celluloid/lib/celluloid/properties.rb:10: warning: method redefined; discarding old finalizer
/home/travis/build/celluloid/celluloid/lib/celluloid/properties.rb:10: warning: previous definition of finalizer was here
/home/travis/build/celluloid/celluloid/lib/celluloid/properties.rb:10: warning: method redefined; discarding old exit_handler_name
/home/travis/build/celluloid/celluloid/lib/celluloid/properties.rb:10: warning: previous definition of exit_handler_name was here
/home/travis/build/celluloid/celluloid/lib/celluloid.rb:55: warning: method redefined; discarding old trap_exit
/home/travis/build/celluloid/celluloid/lib/celluloid.rb:55: warning: previous definition of trap_exit was here
/home/travis/build/celluloid/celluloid/lib/celluloid.rb:59: warning: method redefined; discarding old exclusive
/home/travis/build/celluloid/celluloid/lib/celluloid.rb:59: warning: previous definition of exclusive was here
```

**Comment by digitalextremist** [comment]
@e2's revisions are now in the `reduce-rspec-warnings` branch for revisions before a pull into `master`

**Comment by digitalextremist** [comment]
Resolved in #531 and others.
